main.ts

```diff
@@ -85,9 +82,81 @@ class SettingTab extends PluginSettingTab {
 					.setPlaceholder("Enter your API key")
 					.setValue(this.plugin.settings.openAIApiKey)
 					.onChange(async (value) => {
-						this.plugin.settings.openAIApiKey = value;
-						await this.plugin.saveSettings();
+						await this.plugin.updateSettings({openAIApiKey: value});
+					}),
+			);
+		new Setting(containerEl)
+			.setName("Minimum Summary Size")
+			.setDesc("Minimum number key points to include in generated summary.")
+			.addText((text) =>
+				text
+					.setPlaceholder("3")
+					.setValue("" + this.plugin.settings.summarySize)
+					.onChange(async (value) => {
+						let parsed = parseInt(value);
+						if (isNaN(parsed)) return;
+						parsed = parsed > MAX_SUMMARY_SIZE ? MAX_SUMMARY_SIZE : 
+								 parsed < DEFAULT_SUMMARY_SIZE ? DEFAULT_SUMMARY_SIZE : parsed;
+						await this.plugin.updateSettings({summarySize: parsed});
+					})
+			);
+		new Setting(containerEl)
+			.setName("Maximum Token Size")
+			.setDesc("Maximum size of token in a given request to OpenAI.")
+			.addText((text) =>
+				text
+					.setPlaceholder("2500")
+					.setValue("" + this.plugin.settings.maxTokenSize)
+					.onChange(async (value) => {
+						let parsed = parseInt(value);
+						if (isNaN(parsed)) return;
+						parsed = parsed > MAX_TOKEN_SIZE_IN_A_REQUEST ? MAX_TOKEN_SIZE_IN_A_REQUEST : parsed;
+						await this.plugin.updateSettings({maxTokenSize: parsed});
+					})
+			);
+		new Setting(containerEl)
+			.setName("Template format")
+			.setDesc(
+				"Enter format of template to be used when inserting summary.  Supported fields are {{Date}}, {{Title}}, {{ImageURL}}, {{Description}}, and {{PodcastURL}}.",
+			)
+			.addTextArea((text) => 
+				text
+					.setPlaceholder(DEFAULT_TEMPLATE)
+					.setValue(this.plugin.settings.templateFormat)
+					.onChange(async (value) => {
+						await this.plugin.updateSettings({templateFormat: value});
 					}),
 			);
+		new Setting(containerEl)
+			.setName("Date format")
+            .setDesc("The default date format.")
+			.addTextArea((text) => 
+				text
+					.setPlaceholder(DEFAULT_DATE_FORMAT)
+					.setValue(this.plugin.settings.dateFormat)
+					.onChange(async (value) => {
+						await this.plugin.updateSettings({dateFormat: value});
+					}),
+			);
+		new Setting(containerEl)
+			.setName("Reset defaults")
+			.setDesc(
+				`This will reset to the default settings (without removing your API key).
+				The settings panel will be closed after pressing.
+				`
+				)
+			.addButton((cb) => {
+			  cb.setWarning()
+				.setButtonText("Reset defaults")
+				.onClick(() => {
+					this.plugin.settings.summarySize = DEFAULT_SUMMARY_SIZE;
+					this.plugin.settings.maxTokenSize = MAX_TOKEN_SIZE_IN_A_REQUEST;
+					this.plugin.settings.templateFormat = DEFAULT_TEMPLATE;
+					this.plugin.settings.dateFormat = DEFAULT_DATE_FORMAT;
+				  	this.plugin.saveSettings();
+					this.plugin.unload();
+					this.plugin.load();
+				});
+			});
```

Comment:
Reference for this default reset button: https://github.com/coddingtonbear/obsidian-local-rest-api/blob/main/src/main.ts#L370